synthetic_data/marginal_dist.py

```diff
@@ -58,40 +81,32 @@ def _detect_dist_discrete(col_stats):
         dist (dict): Dictionary stating distribution type along with other parameters for the distribution.
     """
     # Convert strings to ints, ideally DataProfiler should be handling this
-    categories = [int(category) for category in col_stats["categories"]]
-    categorical_count = {int(k): v for k, v in col_stats["categorical_count"].items()}
+    categories = col_stats["categories"]
 
-    categories.sort()
-    observed_freq = [categorical_count[category] for category in categories]
-    sample_size = col_stats["sample_size"]
+    # can sort if all numeric here
+    try:
+        mapping_order = np.argsort(list(map(int, categories)))
+    except:
+        mapping_order = list(range(len(categories)))
 
-    # Shift categories so they start at 0
-    categories = list(range(len(categories)))
+    category_mapping = {i: categories[i] for i in mapping_order}
+    observed_freq = [col_stats["categorical_count"][categories[i]] for i in mapping_order]
+    p = np.array(observed_freq) / sum(observed_freq)
```

Comment:
Ensures proper order of categorical values when categories themselves numeric. Otherwise, distribution mismatch when pulling for a given category.